### ISSUE TITLE
boolean Process variable reading based on JDocument

### ISSUE THREAD

**Issue by ankur-chrungoo** [americanexpress/unify-flowret#18]
I created a workflow definition file with a boolean process variable like this:
"process_variables": [
      {
        "name": "enrich",
        "type": "boolean",
        "value": "true"
      }

The documentation says that while reading/writing the type is automatically converted to the given type.
However, when reading this, I am facing an issue:

This fails with a UnifyException:
            Boolean bEnrich = document.getBoolean("$.process_variables[name=enrich].value");

This works (as a workaround):
            String enrich = document.getString("$.process_variables[name=enrich].value");

I originally tried defining the process variable like this, but this fails (is NOT allowed) in starting a flowret case, since we have to define the value in string format.
"process_variables": [
      {
        "name": "enrich",
        "type": "boolean",
        "value": true
      }

So, how can I define a **boolean** process variable and read it as a boolean itself, and not as a string?
(I am reading the File system json file outside of Flowret steps, so have to read the process variables this way)

**Comment by deepakarora3** [comment]
Hi,

You never have to be reading or writing process variables directly using JDocs. Flowret will do that internally.

The question you have raised is to do with reading a JSON file using JDocs and does not have anything to do with Flowret. You can read more about JDocs here -> https://github.com/americanexpress/unify-jdocs.

To answer your question, if you store value as "true" then as per JSON format it is treated as a string. If you want to store it as boolean you have to store it as true without double quotes. Hence while reading with JDocs, use getString if it is a string or use getBoolean if it is a boolean.

From a Flowret perspective, you always needs to be storing values as strings. So if it is a boolean type, store it as "true" etc.

Hope this clarifies.

**Comment by ankur-chrungoo** [comment]
Yes, that's fine. Our requirement needs this mechanism (outside of flowret boundary but for the workflow of the application), so will use it accordingly.
